models/cluster.x-k8s.io.machine.js

```diff
@@ -7,7 +7,44 @@ import { insertAt, findBy } from '@/utils/array';
 import { get } from '@/utils/object';
 import { downloadUrl } from '@/utils/download';
 import SteveModel from '@/plugins/steve/steve-class';
+// Prevent scaling down control plane or etcd nodes to zero
+// This is a little overly optimised but avoids iterating over the whole machine set every time
```

Comment:
Personally, I would prefer JSDoc style comments to provide a little more context about our parameters and what we're returning. I find the function name of `noOnlyOfRole` to be a bit unclear, especially when `type` is used so much throughout the function body.